### PR TITLE
My branch

### PR DESCRIPTION
**Pull request by GonGustavo**

### CODE REVIEW

**Review by dantzlerdc — CHANGES_REQUESTED** [review]
check (jacoco-check) on project UnitTestAssignment: Coverage checks have not been met. See log for details. 
The build failed due to jacoco not meeting the coverage requirement

File: src/main/java/StringManipulation.java [hunk 1]
```diff
@@ -1,33 +1,97 @@
...
+
+
     @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        String[] words = string.split("\\s+");
+        int numWords = words.length;
+
+        if (startWord <= 0 || endWord <= 0 || startWord > endWord) {
+            throw new IllegalArgumentException("Invalid start or end word");
```
Comment (dantzlerdc) [inline src/main/java/StringManipulation.java:61]: I ran the test coverage on my local machine to see what test coverage issues jacoco is complaining about.
I was not able to upload a screen shot, but basically it identifies those lines that throw an exception as not being accounted for. Check you test cases to see if you check for those exceptions.

File: src/test/java/StringManipulationTest.java [hunk 2]
```diff
@@ -55,27 +63,32 @@ public void testRemoveNthCharacter2() {
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Bring");
+        assertEquals("Bri", manipulatedstring.removeNthCharacter(5, false));
```
Comment (dantzlerdc) [inline src/test/java/StringManipulationTest.java:73]: You are missing test cases to check for Exceptions being thrown. For example
 @Test //assert throws IllegalArgumentException if n == 0
    public void testRemoveNthCharacter3() {
        manipulatedstring.setString("123");
        assertThrows(IllegalArgumentException.class, () -> {manipulatedstring.removeNthCharacter(0,true);});
    }

**Review by dantzlerdc — CHANGES_REQUESTED** [review]
You need to add test cases to cover the possible Exceptions that can be thrown.
For example:
 @Test //assert throws IllegalArgumentException if n == 0
    public void testRemoveNthCharacter3() {
        manipulatedstring.setString("123");
        assertThrows(IllegalArgumentException.class, () -> {manipulatedstring.removeNthCharacter(0,true);});
    }

**Comment by GonGustavo** [comment]
Thank you Dewayne, after reviewing your feedback I was able to correct my code and get it to pass in Travis.

**Comment by dantzlerdc** [comment]
Great, then I will go ahead a close this ticket.

**Pull request closed without merging**